UOP1_Project/Assets/Scripts/UI/FadeManager.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class FadeManager : MonoBehaviour
+{
+	[Header("Fadeout Event")]
+	[SerializeField] private FadeChannelSO _fadeChannelSO = default;
+
+	[Header("Fadeout Canvas")]
+	public GameObject _fadeoutCanvas = default;
+
+	private Image _imageComponent;
+
+	private bool IsCurrentlyFading = false;
+
+	/// <summary>
+	/// Enumerators that fades in the canvas's imageComponent to turn the screen to a flat color over time. Fadeins called simeutaneously will only fade in the earliest call and discard any others.
+	/// </summary>
+	/// <param name="duration">How long it takes to the image to fade in.</param>
+	/// <param name="color">Target color for the image to reach.</param>
+	/// <returns></returns>
+	private IEnumerator FadeInEnum(float duration, Color color)
+	{
+		float totalTime = 0f; // Total amount of time this coroutine has taken. Determines when the fadein will end and what color the imageComponent should be at every frame.
+		while (totalTime <= duration)
+		{
+			totalTime += Time.deltaTime;
+			_imageComponent.color = Color.Lerp(new Color(0, 0, 0, 0), color, totalTime/duration); // Sets the image's color to a mixture between total transparency and the target color, and interpolates based on the amount of time to completion.
+			yield return null;
+		}
+		_imageComponent.color = color; // Here to guarentee the image is exactly the requested color at the end of the loop.
+		IsCurrentlyFading = false;
+		yield return null;
+	}
+
+	/// <summary>
+	/// Enumerators that fades out the canvas's imageComponent to turn the screen to normal gameplay color over time. Fadeouts called simeutaneously will only fade out the earliest call and discard any others.
+	/// </summary>
+	/// <param name="duration">How long it takes to the image to fade out.</param>
+	/// <returns></returns>
+	private IEnumerator FadeOutEnum(float duration)
+	{
+		Color oldColor = _imageComponent.color; // Temporarily stores the old color of the image component, as we can't assume the image will always be black.
+		float totalTime = 0f; // Total amount of time this coroutine has taken. Determines when the fadeout will end and what color the imageComponent should be at every frame.
+		while (totalTime <= duration)
+		{
+			totalTime += Time.deltaTime;
+			_imageComponent.color = Color.Lerp(oldColor, new Color(0, 0, 0, 0), totalTime / duration); // Sets the image's color to a mixture between the old color and  total transparency, and interpolates based on the amount of time to completion.
+			yield return null;
+		}
+		_imageComponent.color = new Color(0, 0, 0, 0); // Here to guarentee the image is fully transparent at the end of the loop.
+		IsCurrentlyFading = false;
+		yield return null;
+	}
```

Comment:
Here I said [in the Thread](https://forum.unity.com/threads/ui-fader-to-black.1038013/#post-6728329) that I would like to keep the ability to choose a FadeOut starting color

you could do this with a default value so that if you don't put anything as a parameter it automatically chooses the old color:
```suggestion
	private IEnumerator FadeOutEnum(float duration, Color color = default)
	{
		if (color == default) color = _imageComponent.color; // Stores the old color of the image component, as we can't assume the image will always be black, if no color is specified.
		float totalTime = 0f; // Total amount of time this coroutine has taken. Determines when the fadeout will end and what color the imageComponent should be at every frame.
		while (totalTime <= duration)
		{
			totalTime += Time.deltaTime;
			_imageComponent.color = Color.Lerp(color, new Color(0, 0, 0, 0), totalTime / duration); // Sets the image's color to a mixture between the old color and  total transparency, and interpolates based on the amount of time to completion.
			yield return null;
		}
		_imageComponent.color = new Color(0, 0, 0, 0); // Here to guarentee the image is fully transparent at the end of the loop.
		IsCurrentlyFading = false;
		yield return null;
	}
```

Reply:
You would need to change it in the FadeChannelSO.cs as well so that You can choose a FadeOut starting color there as well.

Reply:
and you'd need to change it in this file line 96 `StartCoroutine(FadeOutEnum(duration));` aswell

Reply:
Your two comments make sense. I'll implement the changes and update the PR.